packages/nextjs/components/scaffold-eth/Contract/utilsComponents.tsx

```diff
@@ -1,60 +1,81 @@
-import { BigNumber } from "ethers";
-import { Dispatch, SetStateAction, useState } from "react";
+import { Dispatch, SetStateAction } from "react";
 import { toast } from "~~/utils/scaffold-eth";
 
+const NUMBER_REGEX = /^\d+\.?\d*$/;
+
 /**
- * Input component which comes with utility buttons to handle conversion of ETH
+ * Removes redundant leading and trailing zeros from stringified number
+ * @param num stringified number
+ */
+const removeRedundantZeroes = (num: string) => {
+  return (
+    num
+      // leading
+      .replace(/^0+(0\.|[1-9]+)/, "$1")
+      // trailing
+      .replace(/(\.[0-9]*[1-9])0+$|\.0*$/, "$1")
+  );
+};
+
+/**
+ * Multiplies stringified number to positive power of 10
+ * @param num stringified number
+ * @param powerOf10 positive power of 10
+ * @returns {string} result of multiplying
  */
-const TxValueInput = ({
-  setTxValue,
-  txValue,
-}: {
-  setTxValue: Dispatch<SetStateAction<BigNumber | undefined>>;
-  txValue: BigNumber | undefined;
-}) => {
-  const [isHex, setIsHex] = useState(false);
-  const inputValue = txValue
-    ? isHex
-      ? BigNumber.from(txValue).toHexString()
-      : BigNumber.from(txValue).toString()
-    : "";
+const multiplyStringifiedNumberToPowerOf10 = (num: string, powerOf10: number): string => {
+  if (!num || !NUMBER_REGEX.test(num)) {
+    return "";
+  }
+  const numberParts = num.split(".");
+  let result = "";
+  if (numberParts.length === 1) {
+    result = `${num}${"0".repeat(powerOf10)}`;
+  } else if (numberParts[1] && powerOf10 < numberParts[1].length) {
+    result = `${numberParts[0]}${numberParts[1].slice(0, powerOf10)}.${numberParts[1].slice(powerOf10)}`;
+  } else {
+    result = `${numberParts[0]}${numberParts[1]}${"0".repeat(powerOf10 - numberParts[1].length)}`;
+  }
+
+  return removeRedundantZeroes(result);
+};
```

Comment:
Hey @rin-st ,

I'm trying to understand the benefits of adding this complexity 

VS

Converting the string to float + doing 10 ** 18 + back to string as it was before. Specially now that we won't have this issue anymore: https://github.com/scaffold-eth/se-2/pull/101 

What do you think?

Reply:
https://user-images.githubusercontent.com/25638585/214067007-0dd09909-d892-41f9-ae13-7f903b547d7e.mov

1) JS doesn't always correctly calculates values, especially big numbers
2) you can't send tx when tx value is something like 1e+36

Reply:
I appreciate your thoroughness ;) Even if not super important in this page.

Added some comments to `multiplyStringifiedNumberToPowerOf10` (it's not very friendly to read :D) and changed the error handling a bit (too many toast XD).

Good job.